internal/command/views/hook_ui.go

```diff
@@ -323,11 +323,19 @@ func (h *UiHook) PostImportState(id terraform.HookResourceIdentity, imported []p
 	return terraform.HookActionContinue, nil
 }
 
-func (h *UiHook) PrePlanImport(id terraform.HookResourceIdentity, importID string) (terraform.HookAction, error) {
-	h.println(fmt.Sprintf(
-		h.view.colorize.Color("[reset][bold]%s: Preparing import... [id=%s]"),
-		id.Addr, importID,
-	))
+func (h *UiHook) PrePlanImport(id terraform.HookResourceIdentity, importTarget cty.Value) (terraform.HookAction, error) {
+	if importTarget.Type().IsObjectType() {
+		h.println(fmt.Sprintf(
+			h.view.colorize.Color("[reset][bold]%s: Preparing import... [identity=%s]"),
+			id.Addr, importTarget.GoString(), // TODO improve
```

Comment:
Has Terraform a helper method somewhere to nicely display a cty.Object?

Reply:
I don't know if there's anything applicable for this situation, since any of the object rendering code is going to be for multi-line output. This probably needs something more specialized for the hook output